ERCS/erc-7529.md

```diff
@@ -0,0 +1,157 @@
+
+It is RECOMMENDED that EVM address strings adhere to [ERC-1191](./erc-1191.md) so that the browser client can checksum the validity of the address and its target network before making an RPC call. 
+
+A user can check the propagation of their TXT record from the browser console with `fetch`:
+
+```
+await fetch("https://cloudflare-dns.com/dns-query?name=ERC-7529.1._domaincontracts.example.com&type=TXT", {
+  headers: {
+    Accept: "application/dns-json"
+  }
+})
+```
+
+### Smart Contract Association with a Domain 
+
+Any smart contract MAY implement this ERC to provide a verification mechanism of smart contract addresses listed in a compatible TXT record.
+
+A smart contract need only store one new member variable, `domains`, which is an array of all unique eTLD+1 domains associated with the contract. This member variable can be written to with the external functions `addDomain` and `removeDomain`.
+
+```solidity
+{
+  public string[] domains; // a string list of eTLD+1 domains associated with this contract
+
+  function addDomain(string memory domain) external; // an authenticated method to add an eTLD+1
+
+  function removeDomain(string memory domain) external; // an authenticated method to remove an eTLD+1
+}
+```
+
+### Client-side Verification
+
+The user client MUST verify that the eTLD+1 of the TXT record matches an entry in the `domains` list of the smart contract.
+
+When a client detects a compatible TXT record listed on an eTLD+1, it MUST loop through each listed contract address and, via an appropriate RPC provider, collect the `domains` array from each contract in the list. The client should detect an eTLD+1 entry in the contract's `domains` array that exactly matches (DNS domains are not case-sensitive) the eTLD+1 of the TXT record. 
+
+Alternatively, if a client is inspecting a contract that implements this ERC, the client SHOULD collect the `domains` array from the contract and then attempt to fetch TXT records from all listed eTLD+1 domains to ascertain its association or authenticity. The client MUST confirm that the contract address is contained in a TXT record's `VALUE` field of the eTLD+1 pointed to by the contract's `domains` array. 
+
+## Rationale
+
+According to Cloudflare, the two most common use cases of TXT records today are email spam prevention (via [SPF](https://www.rfc-editor.org/rfc/rfc7208), DKIM, and [DMARC](https://www.rfc-editor.org/rfc/rfc7489) TXT records) and domain ownership verification. The use case considered here for on-chain smart contract discovery and verification is essentially analogous. Now that DoH is supported by most major DNS providers, it is easy to leverage TXT records directly in wallets, dApps, and other web applications without relying on a proprietary vendor API to provide DNS information and thus gain the same benefits afforded to email and domain verification with digital assets and on-chain logic. 
```

Comment:
This paragraph belongs in the motivation section. The motivation justifies the proposal as a whole, while the rationale section should explain individual technical choices made within the proposal itself.